src/EditorFeatures/Core/Implementation/AutomaticCompletion/Sessions/EditorBraceCompletionSession.cs

```diff
@@ -0,0 +1,151 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.BraceCompletion;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.CodeAnalysis.Text.Shared.Extensions;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.BraceCompletion;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Operations;
+using Microsoft.VisualStudio.Utilities.Internal;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.AutomaticCompletion.Sessions
+{
+    internal class EditorBraceCompletionSession : IEditorBraceCompletionSession
+    {
+        private readonly IBraceCompletionService _braceCompletionService;
+        private readonly IEditorOperationsFactoryService _editorOperationsFactoryService;
+        private readonly ISmartIndentationService _smartIndentationService;
+        private readonly ITextUndoHistoryRegistry _undoHistoryRegistry;
+
+        public EditorBraceCompletionSession(
+            IBraceCompletionService braceCompletionService,
+            IEditorOperationsFactoryService editorOperationsFactoryService,
+            ISmartIndentationService smartIndentationService,
+            ITextUndoHistoryRegistry undoHistoryRegistry)
+        {
+            _braceCompletionService = braceCompletionService;
+            _editorOperationsFactoryService = editorOperationsFactoryService;
+            _smartIndentationService = smartIndentationService;
+            _undoHistoryRegistry = undoHistoryRegistry;
+        }
+
+        public virtual void AfterReturn(IBraceCompletionSession session, CancellationToken cancellationToken)
```

Comment:
this is code originally from CurlyBraceCompletionSession.AfterReturn, but heavily modified to generically apply any IBraceCompletionService.GetTextChangesAfterReturn

Currently the only implementation though is still CurlyBraceCompletionService